include/posix/sys/types.h

```diff
@@ -76,6 +76,39 @@ typedef struct pthread_rwlock_obj {
 	k_tid_t wr_owner;
 } pthread_rwlock_t;
 
+typedef u32_t pthread_once_t;
+
+/* pthread_key */
+typedef void *pthread_key_t;
+
+typedef struct pthread_key_obj {
+	/* Optional destructor that is passed to pthread_key_create() */
+	void (*destructor)(void *);
+
+	/* List of KeyData objects that contain thread
+	 * specific data for the key
+	 */
+	sys_dlist_t KeyDataL;
+} pthread_key_obj;
+
+typedef struct ThreadData {
```

Comment:
This is an implementation detail, and as such, this should be declared in the implementation file only.  The rationale is to not pollute the namespace of any file including `sys/types.h` with things that are not there by the standard.  Also, please pay attention to the naming scheme used throughout the project; use `unix_hacker_style` instead of `CamelCaseStyle`.

Reply:
Okay, I have moved both the `thread_data` and `key_data` structs to the implementation file `pthread_key.c`, but retained the `pthread_key_obj` struct in the `sys/types.h` file only.

I have also renamed these (to follow `unix_hacker_style` convention) to start with `pthread_` so as to avoid conflicts with the corresponding type structure objects declared and used in the pthread_key APIs.